internal/discovery/host.go

```diff
@@ -170,12 +170,12 @@ func getOSVersion() string {
 	return infoStat.PlatformVersion
 }
 
-func getTotalMemoryMB() int {
+func getTotalMemoryMB() uint64 {
```

Comment:
This change is needed to fix the lint error
```
G115: integer overflow conversion uint64 -> int64
```

The change is considered safe because:
1. We only support 64-bit architectures
2. There is no negative memory byte count

Alternatively, to leave the signature untouched we would have to prove that the runtime value is inside the `int` range for the running architecture; is not trivial to decide what to do if it does not (panic? return default?)